Recipe for AssaultCube 1.3.0.2

Peppersawce opened the pull request:
Kind of a hackjob but it runs more or less ok on 64bit. Server binary seems to work, online multiplayer and 32bit untested.

Had to comment out a line at startup or else the binary started spewing bs about "OpenGL not being present" or something.

Game data is in the same dir as the binaries because the code is barely readable (for me). I did manage to declare the correct settings dir though.

Begasus commented on games-fps/assaultcube/assaultcube-1.3.0.2.recipe:
```diff
@@ -0,0 +1,93 @@
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	AssaultCube$secondaryArchSuffix = $portVersion
```
Lower case for portName here also

Begasus commented on games-fps/assaultcube/assaultcube-1.3.0.2.recipe:
```diff
@@ -0,0 +1,93 @@
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libgl$secondaryArchSuffix
```
Nitpicking :) `lib:libGL$secondaryArchSuffix`

Peppersawce replied:
I'll do it but all this lowercase/uppercase is kind of sort of really confusing ngl.

Begasus commented:
This goes straight into full-screen (which I don't like - my 2 cents), but I get a black screen with the navigation compass, see the shots fired, but that's about it?

Peppersawce commented:
> This goes straight into full-screen (which I don't like - my 2 cents), but I get a black screen with the navigation compass, see the shots fired, but that's about it?

Press 'Esc' :)

DigitalBox98 commented:
To force the windowed mode, I think it's just this file to put in the package : 
/boot/system/apps/AssaultCube/config/autostart/0.cfg
or 
/boot/home/config/settings/assaultcube/config/autostart/0.cfg

As mentionned by Peppersawce in a private message, the below line set the game in windowed mode : 
fullscreen 0; screenres 1024 768; resetgl;

Peppersawce commented:
I'm not sure it's a good idea to put a config file inside the package for that, it creates the opposite problem, like, people wanting to run it fullscreen but for some reason that ends up not  working.

If you guys feel like the whole deal needs some explanation or something I guess I could add a txt file to explain it in the package's documentation.

Peppersawce commented:
After checking how the game is packaged on some other systems I discovered some more weirdness to it:
* The client will save video settings and apply them on the fly if it starts with the "--init" command (for some reason), this also prevents the "fullscreen to windowed" transition the binary from the older recipe suffered from.
* On other systems the client will load a map at startup instead of starting inside a black room (without specifying any extra command)

Seeing this I removed "HowToWindowed" and added a script that starts the game specifying both "--init" and the starting map used on the other systems. I don't know why that map is loaded automatically on other systems but... eh, the script works well enough.

Peppersawce commented:
...both builds failed, I guess the recipe relies on some changes in Haiku not present in Beta 5 as it compiles without errors on my setup.
